heroic-component/src/test/java/com/spotify/heroic/metric/QueryTraceTest.java

```diff
@@ -0,0 +1,113 @@
+package com.spotify.heroic.metric;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class QueryTraceTest {
+    @Test
+    public void passiveJson() throws Exception {
+        final ObjectMapper m = new ObjectMapper();
+        final String json = "null";
+        final QueryTrace trace = QueryTrace.PASSIVE;
+
+        assertEquals(json, m.writeValueAsString(trace));
+        assertEquals(trace, m.readValue(json, QueryTrace.class));
+    }
+
+    @Test
+    public void activeJson() throws Exception {
+        final ObjectMapper m = new ObjectMapper();
+        final String json = "{\"what\":{\"name\":\"foo\"},\"elapsed\":42,\"children\":[]}";
+        final QueryTrace trace =
+            new QueryTrace.ActiveTrace(QueryTrace.identifier("foo"), 42L, ImmutableList.of());
+
+        assertEquals(json, m.writeValueAsString(trace));
+        assertEquals(trace, m.readValue(json, QueryTrace.class));
+    }
+
+    @Test
+    public void activeJsonDefault() throws Exception {
+        final ObjectMapper m = new ObjectMapper();
+        final String json = "{\"what\":{\"name\":\"foo\"},\"elapsed\":42,\"children\":[]}";
+        final String serializedJson =
+            "{\"what\":{\"name\":\"foo\"},\"elapsed\":42,\"children\":[]}";
+        final QueryTrace trace =
+            new QueryTrace.ActiveTrace(QueryTrace.identifier("foo"), 42L, ImmutableList.of());
+
+        assertEquals(serializedJson, m.writeValueAsString(trace));
+        assertEquals(trace, m.readValue(json, QueryTrace.class));
+    }
+
+    @Test
+    public void namedWatch() {
+        final QueryTrace.Identifier identifier = QueryTrace.identifier("foo");
+        final QueryTrace.NamedWatch watch =
+            new QueryTrace.ActiveNamedWatch(identifier, Stopwatch.createStarted());
+        final QueryTrace trace = watch.end();
+
+        assertTrue(trace instanceof QueryTrace.ActiveTrace);
+        final QueryTrace.ActiveTrace activeTrace = (QueryTrace.ActiveTrace) trace;
+        assertEquals(identifier, activeTrace.getWhat());
```

Comment:
The Matchers are in `heroic-dist` and not accessible to this module. There is a future refactoring planned to modularize better so that hey can be shared.